client/src/featureform/register.py

```diff
@@ -571,6 +573,67 @@ def register_file(
             self.__registrar, name, owner, self.name(), path, variant, description
         )
 
+    def register_directory(
+        self,
+        name,
+        path,
+        description="",
+        variant: str = "",
+        owner="",
+        tags: List[str] = [],
+        properties: dict = {},
+    ):
+        """Register a directory.
+        When registering a directory, files can be interacted with as a table with columns "filename" and "body".
+        For example:
+        filename                  |            body
+        --------------------------|---------------------------------------
+        featureform_docs.txt      | Featureform allows data scientists....
+        featureform_home.txt      | The Open-Source Virtual Feature Store....
+
+
+        **Examples**:
+        ```
+        pages = local.register_directory(
+            name="scraped_pages",
+            description="A directory of scraped web pages",
+            path="scraper/"
+        )
+        ```
+        Args:
+            name (str): Name for how to reference the directory
+            description (str): Description of the directory
+            path (str): Path to directory
+            variant (str): Directory variant
+            owner (str): Owner of the file
+
+        Returns:
+            source (LocalSource): source
+        """
+        path = os.path.abspath(path)
+        if not os.path.isdir(path):
+            raise Exception(f"Path {path} is not a directory")
+
+        if owner == "":
+            owner = self.__registrar.must_get_default_owner()
+        if variant == "":
+            variant = self.__registrar.get_run()
+        # Store the file as a source
+        self.__registrar.register_primary_data(
+            name=name,
+            variant=variant,
+            location=Directory(path),
+            is_directory=True,
```

Comment:
I feel as though instead of adding a "is_directory" one could just check if location is a directory

Reply:
but minor